Sources/SpeziOnboarding/ConsentView/ConsentView.swift

```diff
@@ -0,0 +1,405 @@
+/// or a more customized ``ConsentView/init(contentView:actionView:)`` initializer with a custom-provided content and action view.
+///
+/// The ``ConsentView`` is able to automatically export the signed consent form (in the Markdown version) as a PDF  to the Spezi `Standard`. To handle the exported form, the `Standard` needs to conform to ``OnboardingConstraint``. Developers can opt-out by setting the `exportConsentForm` parameter in ``ConsentView/init(header:asyncMarkdown:footer:action:givenNameField:familyNameField:exportConsentForm:)`` to `false`.
+///
+/// ```swift
+/// ConsentView(
+///     asyncMarkdown: {
+///         Data("This is a *markdown* **example**".utf8)
+///     },
+///     action: {
+///         // The action that should be performed once the user has provided their consent.
+///     },
+///     exportConsentForm: false  // Opt-out of the automatic export of the signed consent form as PDF to the Spezi Standard
+/// )
+/// ```
+public struct ConsentView: View {
+    public enum LocalizationDefaults {
+        public static var givenName: FieldLocalizationResource {
+            FieldLocalizationResource(
+                title: LocalizedStringResource("NAME_FIELD_GIVEN_NAME_TITLE", bundle: .atURL(from: .module)),
+                placeholder: LocalizedStringResource("NAME_FIELD_GIVEN_NAME_PLACEHOLDER", bundle: .atURL(from: .module))
+            )
+        }
+        public static var familyName: FieldLocalizationResource {
+            FieldLocalizationResource(
+                title: LocalizedStringResource("NAME_FIELD_FAMILY_NAME_TITLE", bundle: .atURL(from: .module)),
+                placeholder: LocalizedStringResource("NAME_FIELD_FAMILY_NAME_PLACEHOLDER", bundle: .atURL(from: .module))
+            )
+        }
+        public static var exportedConsentFormTitle: LocalizedStringResource {
+            LocalizedStringResource("CONSENT_TITLE", bundle: .atURL(from: .module))
+        }
+    }
+    
+    /// The ``ExportConfiguration`` enables developers to define the properties of the exported consent form.
+    public struct ExportConfiguration {
+        let consentTitle: LocalizedStringResource
+        let paperSize: PaperSize
+        let includingTimestamp: Bool
+        
+        public init(
+            paperSize: PaperSize = .usLetter,
+            consentTitle: LocalizedStringResource = LocalizationDefaults.exportedConsentFormTitle,
+            includingTimestamp: Bool = true
+        ) {
+            self.paperSize = paperSize
+            self.consentTitle = consentTitle
+            self.includingTimestamp = includingTimestamp
+        }
+    }
+    
+    /// The ``ViewState`` indicates in what state the ``ConsentView`` currently is, especially in regards to the consent export functionality.
+    public enum ViewState: Equatable {
+        case base(SpeziViews.ViewState)
+        case signed
+        case export(share: Bool = false)
+        case exported(PDFDocument)
+    }
```

Comment:
To avoid any naming collisions when we add this to the public API, I would suggest to rename this to `ConsentViewState` and make it a standalone type outside of the `ConsentView`.